pkg/util/constants.go

```diff
@@ -35,29 +35,48 @@ var DefaultAdmissionControllers = []string{
 	"ResourceQuota",
 }
 
-// ServiceClusterIP returns the first IP of the ServiceCIDR
+// ServiceClusterIP returns the first usable IP of the Service CIDR (network + 1) for either IPv4 or IPv6.
 func ServiceClusterIP(serviceCIDR string) (net.IP, error) {
-	ip, _, err := net.ParseCIDR(serviceCIDR)
+	base, _, err := net.ParseCIDR(serviceCIDR)
 	if err != nil {
 		return nil, errors.Wrap(err, "parsing default service cidr")
 	}
-	ip = ip.To4()
-	ip[3]++
-	return ip, nil
+	ip := normalizeIP(base)
+	return addToIP(ip, 1), nil
 }
 
-// DNSIP returns x.x.x.10 of the service CIDR
 func DNSIP(serviceCIDR string) (net.IP, error) {
-	ip, _, err := net.ParseCIDR(serviceCIDR)
+	base, _, err := net.ParseCIDR(serviceCIDR)
 	if err != nil {
 		return nil, errors.Wrap(err, "parsing default service cidr")
 	}
-	ip = ip.To4()
-	ip[3] = 10
-	return ip, nil
+	ip := normalizeIP(base)
+	return addToIP(ip, 10), nil
 }
 
 // AlternateDNS returns a list of alternate names for a domain
 func AlternateDNS(domain string) []string {
 	return []string{"kubernetes.default.svc." + domain, "kubernetes.default.svc", "kubernetes.default", "kubernetes", "localhost"}
 }
+
+// normalizeIP returns a 4-byte slice for v4 or 16-byte slice for v6.
```

Comment:
This function can also return nil. If ip is nil or malformed, ip.To16() can return nil.